utilities/utility.py

```diff
@@ -14,6 +16,35 @@ def update_entity_fields(entity, **kwargs):
     return entity
 
 
+def cascade_soft_delete(parent_model, child, parent_id):
+    @event.listens_for(parent_model, 'after_update')
+    def receive_after_update(mapper, connection, target):
+        listen_for_after_flush_event(target, child, parent_id)
+
+
+def listen_for_after_flush_event(target, child, parent_id):
+    @event.listens_for(db_session, "after_flush", once=True)
+    def receive_after_flush(session, context):
+        if target.state == 'archived':
+            child_model = getattr(api, child)
+            # Check for resource name is due to the improper
```

Comment:
KIndly include your comments immediately after the function definition. Use docstrings https://www.python.org/dev/peps/pep-0008/#block-comments